src/Storages/StorageLoop.cpp

```diff
@@ -16,8 +16,11 @@ namespace DB
             : IStorage(table_id_)
             , inner_storage(std::move(inner_storage_))
     {
-        StorageInMemoryMetadata storage_metadata = inner_storage->getInMemoryMetadata();
-        setInMemoryMetadata(storage_metadata);
+        if (inner_storage)
```

Comment:
`inner_storage == nullptr` seems like a misuse. The code fails gracefully (no exception) but it should throw, imho.